distributed/comm/asyncio_tcp.py

```diff
@@ -666,17 +720,34 @@ def _buffer_append(self, data):
 
     def _buffer_peek(self):
         offset = self._offset
-        buf = self._buffers[0]
-        return buf[offset:] if offset else buf
+        buffers = []
+        size = 0
+        count = 0
+        for b in self._buffers:
+            if offset:
+                b = b[offset:]
+                offset = 0
+            buffers.append(b)
+            size += len(b)
+            count += 1
+            if size > self.SENDMSG_MAX_SIZE or count == self.SENDMSG_MAX_COUNT:
```

Comment:
If `len(b)` is particularly large / `size >> self.SENDMSG_MAX_SIZE`, would we want to trim down `b` and move `offset` accordingly? So we're not sending a potentially huge buffer? Not sure if that's a real concern, or even any more performant.

Reply:
No, there's no cost to passing in large buffers (they just might not be written in a single call). The main thing we want to avoid is doing a bunch of excessive work when prepping the buffer list (or passing too many buffers triggering an OS error). It doesn't make sense to prep a huge list of buffers if only a small amount of them will be actually copied to the tcp write buffer.